core/src/session.rs

```diff
@@ -358,7 +358,7 @@ impl Session {
                 loop {
                     match reader.read_event_into(&mut buf) {
                         Ok(Event::Start(ref element)) => {
-                            current_element = std::str::from_utf8(element)?.to_owned()
+                            std::str::from_utf8(element)?.clone_into(&mut current_element)
```

Comment:
Haven't done Rust development in a while, but oh wow, is this the new idiom? Doesn't make for a whole lot better reading... (not a review request; just a rant 😄)

Reply:
Yeah, it comes from the clippy linting requested by https://rust-lang.github.io/rust-clippy/master/index.html#/assigning_clones

But I agree, it's harder to read (and others think so too - https://github.com/rust-lang/rust-clippy/issues/12778)